sw/device/lib/dif/dif_kmac_unittest.cc

```diff
@@ -0,0 +1,92 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_kmac.h"
+
+#include <string>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace dif_kmac_unittest {
+namespace {
+using ::testing::ElementsAre;
+
+TEST(CustomizationStringTest, Encode) {
+  dif_kmac_customization_string_t cs;
+
+  EXPECT_EQ(dif_kmac_customization_string_init(nullptr, 0, &cs), kDifKmacOk);
+  EXPECT_THAT(std::string(&cs.buffer[0], 2), ElementsAre(1, 0));
+
+  EXPECT_EQ(dif_kmac_customization_string_init("", 0, &cs), kDifKmacOk);
+  EXPECT_THAT(std::string(&cs.buffer[0], 2), ElementsAre(1, 0));
+
+  EXPECT_EQ(dif_kmac_customization_string_init("\x00\x00", 2, &cs), kDifKmacOk);
+  EXPECT_THAT(std::string(&cs.buffer[0], 2), ElementsAre(1, 16));
+  EXPECT_THAT(std::string(&cs.buffer[2], 2), ElementsAre(0, 0));
+
+  EXPECT_EQ(dif_kmac_customization_string_init("SHA-3", 5, &cs), kDifKmacOk);
+  EXPECT_THAT(std::string(&cs.buffer[0], 2), ElementsAre(1, 40));
+  EXPECT_EQ(std::string(&cs.buffer[2], 5), "SHA-3");
```

Comment:
I would actually have gone with
```cpp
EXPECT_THAT(cs.buffer, StrEq("\x01\x28SHA-3"));
```

But tbh I'm willing to just drop it. What you've got is probably "good enough".

Reply:
Thanks @mcy, I'm still figuring out the best way to do certain things in GoogleTest so feedback like this is very useful to me.